### PR TITLE
fix(uap): Fix generators should not run under uap builds

### PR DESCRIPTION
**Pull request by Youssef1313**
See discussion in https://github.com/unoplatform/uno.extensions/pull/1039

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/SourceGenerators/SourceGeneratorHelpers/Helpers/PlatformHelper.cs [hunk 1]
```diff
@@ -15,34 +15,18 @@ public class PlatformHelper
...
-				string.IsNullOrEmpty(context.GetMSBuildPropertyValue("Configuration"))
-				&& string.IsNullOrEmpty(context.GetMSBuildPropertyValue("AssemblyName"))
-				&& string.IsNullOrEmpty(context.GetMSBuildPropertyValue("RootNamespace"));
-
 			// Those two checks are now required since VS 16.9 which enables source generators by default
 			// and the uno targets files are not present for uap targets.
 			var isWindowsRuntimeApplicationOutput = context.Compilation.Options.OutputKind == OutputKind.WindowsRuntimeApplication;
 			var isWindowsRuntimeMetadataOutput = context.Compilation.Options.OutputKind == OutputKind.WindowsRuntimeMetadata;
 
-			var isWinAppSDK = windowsAppSDKWinUI?.Equals("True", StringComparison.OrdinalIgnoreCase) ?? false;
```
Comment (Youssef1313) [inline src/SourceGenerators/SourceGeneratorHelpers/Helpers/PlatformHelper.cs:37]: Generator is removed when `WindowsAppSDKWinUI` is true.

So `isWinAppSDK` is always false. Hence, removed the redundant check.

**Comment by Youssef1313** [comment]
@jeromelaban Can we get this merged to test it for https://github.com/unoplatform/uno.extensions/pull/1039 ?

**Comment by jeromelaban** [comment]
It's likely best to try with the PR nuget packages directly first. Use a local source to validate that the build is working as expected (we may not test all the required scenarios in our template tests).

**Comment by jeromelaban** [comment]
Also, https://github.com/unoplatform/uno.extensions/pull/1039 has two issues, as the events generated code does not seem to use a globalized type somewhere, this current PR may not be enough.

**Comment by Youssef1313** [comment]
@jeromelaban The generator is supposed to generate `global::` everywhere

What is the other failure? The only one I could find in CI is related to running the generator in Windows builds.

> Use a local source to validate that the build is working as expected (we may not test all the required scenarios in our template tests).

I'm not sure what other scenarios you'd like to test other than referencing the package in a project and build (which is what template tests do). Would you be able to do that for me?

**Comment by Youssef1313** [comment]
Also tagging @nickrandolph in case he can download the package from artifacts and test it specifically with his PR #1039

**Comment by nickrandolph** [comment]
I will give it a try tomorrow morning

**Comment by Youssef1313** [comment]
Great. If this failed, please send me a binlog.

File: src/SourceGenerators/Uno.UI.SourceGenerators/Content/Uno.UI.SourceGenerators.props [hunk 2]
```diff
@@ -451,10 +450,9 @@
 			   Text="Uno.UI is using Uno.SourceGenerators" />
 	</Target>
 
-	<Target Name="_RemoveRoslynUnoSourceGeneration" BeforeTargets="CoreCompile" Condition="'$(UnoUIUseRoslynSourceGenerators)'=='false' or '$(WindowsAppSDKWinUI)'=='true'">
-		<!---
-	If the users explicitly disables Roslyn source generation, remove the analyzer item which is automatically added by Nuget.
-	-->
+	<Target Name="_RemoveRoslynUnoSourceGeneration" BeforeTargets="CoreCompile" Condition="'$(UnoUIUseRoslynSourceGenerators)'=='false' or '$(WindowsAppSDKWinUI)'=='true' or $(TargetFramework.StartsWith('uap10.0'))">
```
Comment (Youssef1313) [inline src/SourceGenerators/Uno.UI.SourceGenerators/Content/Uno.UI.SourceGenerators.props:453]:
```suggestion
	<Target Name="_RemoveRoslynUnoSourceGeneration" BeforeTargets="BeforeTargets="CoreCompile;XamlPreCompile"" Condition="'$(UnoUIUseRoslynSourceGenerators)'=='false' or '$(WindowsAppSDKWinUI)'=='true' or $(TargetFramework.StartsWith('uap10.0'))">
```

**Comment by jeromelaban** [comment]
I tested the changes, and the `Uno.UI.SourceGenerators.props` is not imported when building for `uap10.0`, so changing the specialized analyzer removal is enough.

**Review by jeromelaban — APPROVED** [review]

**Pull request merged**